File: JavaFundamentals/JavaAdvanced/SetsAndMaps_Lab/SoftuniParty.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Set;
import java.util.TreeSet;

public class SoftuniParty {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String guest = reader.readLine();
        Set<String> vipGuests = new TreeSet<>();
        Set<String> guests = new TreeSet<>();
        while (!"PARTY".equals(guest)){
            if (Character.isDigit(guest.charAt(0))){
                vipGuests.add(guest);
            } else {
                guests.add(guest);
            }
            guest = reader.readLine();
        }

        guest = reader.readLine();
        while (!"END".equals(guest)){
            if (vipGuests.contains(guest)){
                vipGuests.remove(guest);
            } else {
                guests.remove(guest);
            }
            guest = reader.readLine();
        }
        guests.addAll(vipGuests);
        System.out.println(guests.size());
        for (String missing : guests) {
            System.out.println(missing);
        }
    }
}

File: CSharpFundamentals/CSharpAdvanced/SetsAndDictionaries_Lab/SoftUniParty.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace SoftUniParty
{
    class SoftUniParty
    {
        static void Main(string[] args)
        {
            var vipSet = new HashSet<string>();
            var set = new HashSet<string>();

            string guest = Console.ReadLine();
            while (guest != "PARTY")
            {
                if (char.IsDigit(guest[0]))
                {
                    vipSet.Add(guest);
                }
                else
                {
                    set.Add(guest);
                }

                guest = Console.ReadLine();
            }

            string coming = Console.ReadLine();
            while (coming != "END")
            {
                if (vipSet.Contains(coming))
                {
                    vipSet.Remove(coming);
                }
                else if (set.Contains(coming))
                {
                    set.Remove(coming);
                }

                coming = Console.ReadLine();
            }

            Console.WriteLine(set.Count + vipSet.Count);
            vipSet.ToList().ForEach(s => { Console.WriteLine(s); });
            set.ToList().ForEach(s => { Console.WriteLine(s); });
        }
    }
}
